moz_kinto_publisher/main.py

```diff
@@ -376,14 +381,28 @@ def set_pem(self, pem_data):
         self.pemData = pem_data
         self.pemHash = hashlib.sha256(pem_data.encode("utf-8")).hexdigest()
         derCert = asciiPemToBinaryDer(pem_data)
-        self.derHash = hashlib.sha256(derCert).digest()
         try:
             self.cert = x509.load_pem_x509_certificate(
                 pem_data.encode("utf-8"), default_backend()
             )
         except Exception as e:
             raise IntermediateRecordError("Cannot parse PEM data: {}".format(e))
 
+        derHash = hashlib.sha256(self.cert.public_bytes(Encoding.DER)).digest()
+        if self.derHash and self.derHash != derHash:
+            raise IntermediateRecordError("DER hash does not match")
+        self.derHash = derHash
+
+        self.subject = self.cert.subject.rfc4514_string()
+
+        der_spki = self.cert.public_key().public_bytes(
```

Comment:
nit: it seems like the style in this file would be `derSpki`, but it's not a big deal